pkg/tests/ossm/smcp_annotation_test.go

```diff
@@ -36,99 +40,92 @@ var (
-		} else {
-			util.KubeApplyContents("bookinfo", testenvDeploymentX86)
-		}
-		util.CheckPodRunning("bookinfo", "app=env")
-		msg, err := util.ShellMuteOutput(`kubectl get po -n bookinfo -o yaml | grep maistra_test_env`)
-		util.Inspect(err, "Failed to get variables", "", t)
-
-		if strings.Contains(msg, "env_value") {
-			log.Log.Info(msg)
-		} else {
-			t.Errorf("Failed to get env variable: %v", msg)
-		}
-		util.KubeDeleteContents("bookinfo", testenvDeploymentX86)
-		time.Sleep(time.Duration(30) * time.Second)
+	NewTest(t).Id("T29").Groups(Full).Run(func(t TestHelper) {
+		hack.DisableLogrusForThisTest(t)
+		ns := "foo"
+		t.Cleanup(func() {
+			oc.RecreateNamespace(t, meshNamespace)
+		})
+
+		t.NewSubTest("proxyEnvoy").Run(func(t TestHelper) {
+			t.Cleanup(func() {
+				oc.RecreateNamespace(t, ns)
+			})
+			t.LogStep("Deploy TestSSL pod with annotations sidecar.maistra.io/proxyEnv")
+			DeployTestSsl(t, ns)
+
+			t.LogStep("Get annotations and verify that the pod has the expected ones")
+			annotations := make(map[string]string)
+			annotations["sidecar.maistra.io/proxyEnv"] = `{ "maistra_test_env": "env_value", "maistra_test_env_2": "env_value_2" }`
+			pod := pod.MatchingSelector("app=env", ns)
+			podAnnotations := GetPodAnnotations(t, pod)
+			for k, v := range annotations {
+				t.Logf("Checking annotation %s=%s", k, v)
+				if podAnnotations[k] != v {
+					t.Fatalf("Expected annotation %s=%s, but got %s", k, v, podAnnotations[k])
+				}
+			}
+		})
+
+		// Test that the SMCP automatic injection with quotes works
+		t.NewSubTest("quote_injection").Run(func(t TestHelper) {
+			t.Cleanup(func() {
+				oc.RecreateNamespace(t, ns)
+			})
+			t.LogStep("Enable annotation auto injection in SMCP")
+			oc.Patch(t, meshNamespace, "smcp", smcpName, "merge", InjectedAnnotationsSMCPPatch)
+			oc.WaitSMCPReady(t, meshNamespace, smcpName)
+
+			t.LogStep("Deploy TestSSL pod with annotations sidecar.maistra.io/proxyEnv")
+			DeployTestSsl(t, ns)
+
+			t.LogStep("Get annotations and verify that the pod has the expected ones")
+			annotations := make(map[string]string)
+			annotations["test1.annotation-from-smcp"] = "test1"
+			annotations["test2.annotation-from-smcp"] = `["test2"]`
+			annotations["test3.annotation-from-smcp"] = "{test3}"
+			pod := pod.MatchingSelector("app=env", ns)
+			podAnnotations := GetPodAnnotations(t, pod)
+			for k, v := range annotations {
+				if podAnnotations[k] != v {
+					t.Fatalf("Expected annotation %s=%s, but got %s", k, v, podAnnotations[k])
+				}
+			}
+		})
+
 	})
+}
+
+func DeployTestSsl(t TestHelper, ns string) {
+	yaml := getTestSsslYaml()
+	oc.ApplyString(t, ns, yaml)
+	operatorPod := pod.MatchingSelector("app=env", ns)
+	oc.WaitPodRunning(t, operatorPod)
+}
+
+func getTestSsslYaml() string {
+	yaml := ""
+	switch env.Getenv("SAMPLEARCH", "x86") {
+	case "p":
+		yaml = testenvDeploymentP
+	case "z":
+		yaml = testenvDeploymentZ
+	default:
+		yaml = testenvDeploymentX86
+	}
+	return yaml
```

Comment:
I think we should convert one of these three manifests into a template, and execute the template like we did here: https://github.com/maistra/maistra-test-tool/blob/maistra-2.4/pkg/tests/ossm/testssl_test.go#L84-L115

Reply:
Done